MaxText/standalone_dataloader.py

```diff
@@ -53,12 +54,18 @@ def data_load_loop(config, state=None):
 
   jax.block_until_ready(example_batch) # wait until the last batch is read
   end = datetime.datetime.now()
-  max_logging.log(f"{config.steps} batches loaded in {end-start} seconds, on host {jax.process_index()}")
+  if jax.process_index() == 0:
+    max_logging.log(f"STANDALONE DATALOADER: {config.steps} batches loaded in {end-start} seconds, on host 0")
   return state
 
 
 def main(argv: Sequence[str]) -> None:
   jax.config.update('jax_default_prng_impl', 'unsafe_rbg')
+  jax.config.update('jax_cpu_enable_gloo_collectives', True)
+  jax.config.update('jax_platforms', 'cpu')
+  jax.distributed.initialize(coordinator_address=socket.gethostbyname(os.environ.get("JAX_COORDINATOR_ADDRESS")),
```

Comment:
@rwitten @gobbleturk I think this line needs to be specific to CPUs (v4 runner tests are failing due to this line)  as these JAX variables will only be set for CPUs.

Is there a check that you could recommend?

Reply:
This is a Matt problem!

Reply:
Sure, will wait for his response, thanks Rafi!

Reply:
Thanks for the offline advise, Matt! @gobbleturk 

I moved the jax distributed initialize for CPUs to max utils. I also tested that async checkpointer now works for CPUs, with this changes.

Logs: 
with async checkpointer - https://cloudlogging.app.goo.gl/V7Qog8rJvRDgJRci8 
with sync checkpointer - https://cloudlogging.app.goo.gl/XzxRKqBYD2tXxW8g8 


Test failures are due to "Install dependencies" step, probably unrelated to my changes.

Reply:
https://github.com/google/maxtext/pull/399 will unblock the tests, I will rebase onto it shortly.